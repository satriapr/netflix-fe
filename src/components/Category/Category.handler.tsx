import { useEffect, useState } from 'react'

import { IMovieReq, IMovieRes } from '../../models'
import movieService from '../../services/movie.service'

// Hooks
const Handler = (props: IMovieReq) => {
  const [movies, setMovies] = useState<IMovieRes[]>([])
  const [isLoading, setIsLoading] = useState<boolean>(true)

  useEffect(() => {
    const getMovies = async () => {
      const response = await movieService.getMovies(props)
      setMovies(response)
      setIsLoading(false)
    }

    getMovies()
  }, [props])

  return { isLoading, movies, setMovies }
}

export default Handler
