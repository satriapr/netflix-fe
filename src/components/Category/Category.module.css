.menuItem {
  padding: 0px;
  margin: 5px 2px;
  user-select: none;
  cursor: pointer;
  border: none;
}

.menuItem.active {
  border: 1px green solid;
}

/*lists of movies*/
.lists {
  margin: 100px 0;
}

.lists:first-child {
  margin-top: 50px;
}

.lists h2 {
  padding-left: 70px;
  margin-bottom: 10px;
}

.categoryTitle {
  padding-bottom: 20px
}

@media only screen and (max-width: 900px) {
  .lists {
    margin: 0;
  }

  .lists h2 {
    margin-bottom: 0;
  }
}

@media only screen and (max-width: 690px) {
  .lists h2 {
    padding-left: 5px;
    margin-bottom: 10px;
  }
}