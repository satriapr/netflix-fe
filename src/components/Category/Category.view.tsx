import React from 'react'
import ScrollMenu from 'react-horizontal-scrolling-menu'

import { IMovieReq, IMovieRes } from '../../models'
import { ArrowLeft, ArrowRight } from '../CategoryArrow/CategoryArrow.view'
import Movie from '../Movie/Movie.view'
import categoryHandler from './Category.handler'
import styles from './Category.module.css'

const Category = (props: IMovieReq) => {
  const handler = categoryHandler(props)
  const { movies, isLoading } = handler

  const menu = movies.map((movie: IMovieRes, i) => {
    const { Title, imdbID, Poster } = movie

    return (
      <div className={styles.menuItem} key={i}>
        <Movie Title={Title} imdbID={imdbID} Poster={Poster} />
      </div>
    )
  })

  return (
    <React.Fragment>
      {!isLoading ? (
        <div className={styles.lists}>
          <h2 className={styles.categoryTitle}>{props.categoryTitle}</h2>
          <ScrollMenu
            data={menu}
            arrowLeft={ArrowLeft}
            arrowRight={ArrowRight}
            dragging={true}
            wheel={false}
            alignCenter={false}
            clickWhenDrag={false}
          />
        </div>
      ) : (
        <div className="loading-mask" />
      )}
    </React.Fragment>
  )
}

export default Category
