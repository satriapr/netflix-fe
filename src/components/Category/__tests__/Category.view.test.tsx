import Category from '../Category.view'
import { testSnapshots } from '../../../utils/test.util'

describe('Category Component', () => {
  testSnapshots(Category, [
    {
      description: 'Category props',
      props: {
        keyword: 'man',
        type: 'movie',
        categoryTitle: 'Popular on Netflix'
      }
    }
  ])
})
