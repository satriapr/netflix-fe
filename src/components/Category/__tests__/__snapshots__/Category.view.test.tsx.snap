// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Category Component snapshots Category props 1`] = `
<Fragment>
  <div
    className="loading-mask"
  />
</Fragment>
`;
